### ISSUE TITLE
Builds randomly fail to pull custom docker image 

### ISSUE THREAD

**Issue by npahucki** [Shippable/support#3787]
We had this problem a few days ago, then it resolved itself (or rebuilding the same build fixed it), but now one of our engineers is in a situation where no matter how many times she tries to rebuild, it simply won't pass. Here is an example: https://app.shippable.com/github/ExecThread/ExecThread/runs/5692/1/console

We have confirmed that ECS is normal, and that the mentioned image CAN be pulled down locally (so it's not likely an issue with ECS itself). 

```
  shippable_retry sudo docker pull 728892293674.dkr.ecr.us-east-1.amazonaws.com/execthread/build-base:latest 5s  
Pulling repository 728892293674.dkr.ecr.us-east-1.amazonaws.com/execthread/build-base
Error: image execthread/build-base:latest not found
retrying 1 of 3 times...
Pulling repository 728892293674.dkr.ecr.us-east-1.amazonaws.com/execthread/build-base
Error: image execthread/build-base:latest not found
retrying 2 of 3 times...
Pulling repository 728892293674.dkr.ecr.us-east-1.amazonaws.com/execthread/build-base
Error: image execthread/build-base:latest not found
retrying 3 of 3 times...
```

**Comment by szarekm** [comment]
We are experiencing the same issue. Problems with pulling docker started about an hour ago.

**Comment by louib** [comment]
Same problem here also. Started maybe an hour ago.

**Comment by ambarish2012** [comment]
we are investigating.

**Comment by ambarish2012** [comment]
I am able to pull a private ECR image from us-east-1 - https://app.shippable.com/github/ambarish2012/ci-push-amazon-ecr/runs/6/1/console.

**Comment by ambarish2012** [comment]
Some customers are however still experiencing this issue as of 3:00pm -
https://app.shippable.com/github/twentybelow/Insight-Demo-Imager/runs/25/1/console

**Comment by trriplejay** [comment]
it looks like this is related to the docker version.  @npahucki @szarekm @louib are all running builds on docker 1.9.  It's possible that some change was made to ECR today that is causing this error when talking to 1.9 clients.

We're still investigating if this is a bug with ECR or a permanent change, or something wrong with our AMI, but in the meantime, you can change the version of docker used by your subscription by navigating to the subscription "nodes" page:
![image](https://user-images.githubusercontent.com/6869398/29590877-f13e6338-8750-11e7-8f33-1e5154436e5f.png)

and pick a different "machine image" from the drop down.
![image](https://user-images.githubusercontent.com/6869398/29590910-14b5b582-8751-11e7-86d8-7bf7f98114f6.png)

5.8.2 is the newest release. You can read about the available machine images in this section of our docs:
http://docs.shippable.com/platform/tutorial/runtime/ami-overview/

In my sample, I was able to reproduce the same error with a 1.9 machine image, but my image was pulled successfully with a 17.06 machine image:

failure:
![image](https://user-images.githubusercontent.com/6869398/29591164-1804862c-8752-11e7-821f-d6a1152b3ee8.png)


success:
![image](https://user-images.githubusercontent.com/6869398/29591211-5228a374-8752-11e7-99a0-5d2595bd5f11.png)

**Comment by louib** [comment]
@trriplejay looks like updating the image to 5.8.2 fixed this. thanks!

**Comment by npahucki** [comment]
Confirmed that switching to 5.8.2 fixes it. Thank you!

**Comment by npahucki** [comment]
Actually, I spoke too soon. It does now pull the image ok with 5.8.2, but it fails at a later step with the new image: 
```
  echo -e 'ECR get-login failed\n\n' 0s  
Unknown options: --no-include-email
ECR get-login failed


/root/efd00584-5dad-443b-aa97-24d776c070cf.sh: line 74: Unknown: command not found

Unknown options: --no-include-email
```

**Comment by Bharath92** [comment]
5.8.2 system machine image uses docker 17.06 which doesn't support the `--email` flag for `docker login`. The aws cli needs `--no-include-email` flag to be compatible with the docker 17.06 login. The `--no-include-email` flag is available in aws cli from 1.11.91 onwards. Please update your custom image with aws cli version 1.11.91 or above.

**Comment by deepikasl** [comment]
Closing as it is resolved. Please re-open if you see it again.
